src/shared/dlt_log.c

```diff
@@ -0,0 +1,352 @@
+/*
+ * SPDX license identifier: MPL-2.0
+ *
+ * Copyright (C) 2022, BMW AG
+ *
+ * This file is part of COVESA Project DLT - Diagnostic Log and Trace.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License (MPL), v. 2.0.
+ * If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * For further information see https://www.covesa.global/.
+ */
+
+/*!
+ * \author
+ * Alexander Wenzel <alexander.aw.wenzel@bmw.de>
+ * Markus Klein <Markus.Klein@esk.fraunhofer.de>
+ * Mikko Rapeli <mikko.rapeli@bmw.de>
+ * Oleg Tropmann <oleg.tropmann@daimler.com>
+ * Daniel Weber <daniel.w.weber@daimler.com>
+ *
+ * \copyright Copyright © 2022 BMW AG. \n
+ * License MPL-2.0: Mozilla Public License version 2.0 http://mozilla.org/MPL/2.0/.
+ *
+ * \file dlt_log.c
+ */
+
+#include "dlt_log.h"
+#include "dlt_common.h"
+#include "dlt_multiple_files.h"
+#include <syslog.h>
+#include <errno.h>
+#include <libgen.h>
+#include <string.h>
+#include <stdlib.h>
+
+/* internal logging parameters */
+static int logging_level = LOG_INFO;
+static char logging_filename[NAME_MAX + 1] = "";
+DltLoggingMode logging_mode = DLT_LOG_TO_CONSOLE;
+FILE *logging_handle = NULL;
+
+//use ohandle as an indicator that multiple files logging is active
+MultipleFilesRingBuffer multiple_files_ring_buffer = {
+        .directory={0},
+        .filename={0},
+        .fileSize=0,
+        .maxSize=0,
+        .filenameTimestampBased=false,
+        .filenameBase={0},
+        .filenameExt={0},
+        .ohandle=-1};
+
+
+void dlt_log_set_filename(const char *filename)
```

Comment:
Why is it really necessary  to move all this code from dlt_common.c to dlt_log.c? These cleanup activities make it hard to review the change it the PRs get really big. Two seperate PRs here: one for the clean up / code moving. And a second one which implements the feature.

Reply:
Of course, the split between dlt_common and dlt_log is not mandatory for the feature. I would remove the movement from dlt_common to dlt_common.
If my feature PR is merged, I will create another PR that contains the split.